src/App.tsx

```diff
@@ -51,6 +52,36 @@ const App = () => {
       .getUiSchema()
       .then((uiSchema) => dispatch(Actions.setUiSchema(uiSchema)));
   }, [schemaService]);
+  useEffect(() => {
+    //try to preserve selection when schemas change
+    const oldSelection = selection;
+    if (!oldSelection) {
+      return;
+    }
+    const newSelectedUISchemaElement =
+      oldSelection && uiSchema
+        ? getCorrespondingElement(oldSelection?.uiSchema, uiSchema)
```

Comment:
`getCorrespondingElement` is meant to be used only on cloned states without structural changes as it works path based. Therefore it will fail in many cases here: Whenever you select an element and insert an element somewhere "before" it in the tree, the new selection will be wrong or missing completely.

Reply:
Replaced it with `findByUUID` method